apps/system/js/quick_settings.js

```diff
@@ -170,32 +170,8 @@
      * @memberof QuickSettings.prototype
      */
     monitorBluetoothChange: function() {
-      var self = this;
-      var btFirstSet = true;
-      SettingsListener.observe('bluetooth.enabled', true, function(value) {
-        // check self.bluetooth.dataset.enabled and value are identical
-        if ((self.bluetooth.dataset.enabled && value) ||
-          (self.bluetooth.dataset.enabled === undefined && !value)) {
-          return;
-        }
-
-        if (value) {
-          self.bluetooth.dataset.enabled = 'true';
-        } else {
-          delete self.bluetooth.dataset.enabled;
-        }
-
-        // Set to the initializing state to block user interaction until the
-        // operation completes. (unless we are being called for the first time,
-        // where Bluetooth is already initialize
-        if (!btFirstSet) {
-          self.bluetooth.dataset.initializing = 'true';
-        }
-        btFirstSet = false;
-
-        self.setAccessibilityAttributes(self.bluetooth, 'bluetoothButton');
-      });
-      window.addEventListener('bluetooth-adapter-added', this);
+      // Bluetooth module is loaded after quicksettings.
+      window.addEventListener('bluetooth-enabled', this);
```

Comment:
Per the previous comment, we should listen to the event of the bluetooth state changes instead, so that the state of the button can be updated directly base on it.